File: CS/DragDropColumns/MyFitColumnHeaderDropTargetFactory/MyFitColumnHeaderDropTargetFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Xpf.Grid;
using DevExpress.Xpf.Core;
using System.Windows;

namespace DragDropColumns
{
    class MyFitColumnHeaderDropTargetFactory : GridDropTargetFactoryBase
    {
        protected override IDropTarget CreateDropTarget(UIElement dropTargetElement)
        {
            return new MyFitColumnHeaderDropTarget(dropTargetElement as GridColumnHeaderBase);
        }
    }
}

File: VB/DragDropColumns/MyFitColumnHeaderDropTargetFactory/MyFitColumnHeaderDropTargetFactory.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.Xpf.Grid
Imports DevExpress.Xpf.Core
Imports System.Windows

Namespace DragDropColumns
	Friend Class MyFitColumnHeaderDropTargetFactory
		Inherits GridDropTargetFactoryBase
		Protected Overrides Function CreateDropTarget(ByVal dropTargetElement As UIElement) As IDropTarget
			Return New MyFitColumnHeaderDropTarget(TryCast(dropTargetElement, GridColumnHeaderBase))
		End Function
	End Class
End Namespace
